api/v1alpha1/custom_package_types.go

```diff
@@ -56,6 +56,7 @@ type ArgoCDPackageSpec struct {
 	ApplicationFile string `json:"applicationFile"`
 	Name            string `json:"name"`
 	Namespace       string `json:"namespace"`
+	Type            string `json:"type"`
```

Comment:
Maybe this is beyond the scope of this PR but since these are CRDs and we clearly only support two types we should use the kubebuilder to set enums up for validation. https://book.kubebuilder.io/reference/markers/crd-validation

Reply:
Oh that's 100% a miss on my part. We should be doing that. I will update it.